ethereum/test/beefy-helpers.js

```diff
@@ -0,0 +1,56 @@
+const { ethers } = require("ethers");
+
+const {
+  createMerkleTree
+} = require("./helpers");
+const { keccakFromHexString } = require("ethereumjs-util");
+
+async function createBeefyValidatorFixture(numberOfValidators) {
+
+  let wallets = [];
+  for (let i = 0; i < numberOfValidators; i++) {
+    const wallet = ethers.Wallet.createRandom();
+    wallets.push(wallet);
+  }
+  const walletsByLeaf = wallets.reduce((accum, wallet) => {
+    const leaf = '0x' + keccakFromHexString(wallet.address).toString('hex')
+    accum[leaf] = wallet
+    return accum
+  }, {})
+
+  wallets = wallets.sort((a, b) => a.address < b.address)
```

Comment:
Why sort the wallets?

Reply:
That should not be needed, validator set on Polkadot is also not sorted by the beefy session key, but rather uses a deterministic sorting (maybe by stash key?). The beefy session key which is used for signing should be unsorted though.

Reply:
For reference: 

> //! Inner nodes are created by concatenating child hashes and hashing again. The implementation
> //! does not perform any sorting of the input data (leaves) nor when inner nodes are created.

https://github.com/paritytech/grandpa-bridge-gadget/pull/225/files#diff-0d5537e8e591fa5d42cb1e23f47fa81950441ce7d9a9eca4cdd4621f08cf02d0R28-R29

Reply:
Also, here is the code that concats and hashes the pairs, without sorting: 

https://github.com/paritytech/grandpa-bridge-gadget/pull/225/files#diff-0d5537e8e591fa5d42cb1e23f47fa81950441ce7d9a9eca4cdd4621f08cf02d0R351-R356

Reply:
ah cool - well, then it shouldn't matter. will leave as is and change if needed when integrating real proofs.